src/main/java/org/spongepowered/api/util/command/args/GenericArguments.java

```diff
@@ -895,16 +901,16 @@ protected PlayerCommandElement(Text key, Game game, boolean returnSource) {
             this.returnSource = returnSource;
         }
 
+        @SuppressWarnings("unchecked")
         @Override
         protected Object parseValue(CommandSource source, CommandArgs args) throws ArgumentParseException {
-            // TODO: Make player name resolution better -- support selectors, etc
             if (!args.hasNext() && this.returnSource) {
                 return tryReturnSource(source, args);
             }
 
             Object state = args.getState();
             try {
-                return super.parseValue(source, args);
+                return Iterables.filter((Iterable<Entity>)super.parseValue(source, args), Predicates.instanceOf(Player.class));
```

Comment:
Changed this to lambdas, but I couldn't use streams because I might be given an `Iterable` (which doesn't support streams) as opposed to a `Collection`.